src/components/cylc/TreeItem.vue

```diff
@@ -117,7 +118,18 @@ export default {
   created () {
     this.$emit('tree-item-created', this)
   },
+  beforeDestroy () {
+    if (Object.prototype.hasOwnProperty.call(this.$refs, 'treeitem')) {
+      this.$refs.treeitem.map((treeitem) => {
+        treeitem.destroy()
```

Comment:
The children / recursive TreeItem elements are not always being destroyed. Not sure if because it's happening too soon/late, or if because we are doing something that is not conventional for Vue components (the circular/recursive pattern is no that common). So we call a custom `destroy` method, which will then call `$destroy(true)` method from Vue. This `$destroy` takes care to trigger the `beforeDestroy -> destroy` lifecycle.